### ISSUE TITLE
How to change the callback function to a static call?

### ISSUE THREAD

**Issue by bruceywave** [bytedeco/javacpp#346]
I have a C++ class which has a callback function, and this function is implemented in JAVA. To improve performance, I want to change this function into JAVA static method.

**Comment by saudet** [comment]
BTW, we can still use JNI manually for special needs like this.

**Comment by bruceywave** [comment]
Don't use your javacpp, and then write your own JNI interface by hand? Does javacpp support static callbacks at present?

**Comment by saudet** [comment]
No, it doesn't, but if you'd like to give it a try, it shouldn't be too hard to implement.

**Comment by bruceywave** [comment]
How can I achieve static callback use InfoMapper? for example：
	`typedef void(*cluster_status_notify_j)(bool);
	typedef void(*recv_data_callback_j)(char data_type, uint32_t plc_id, uint64_t peer_id, char ret_code, uint32_t result_cnt, void* presult);

	struct java_cluster_options {
		cluster_status_notify_j cluster_status_notify_callback;
		recv_data_callback_j pub_data_callback; // called when subscribed data received, used by 3: other
		recv_data_callback_j ack_data_callback; // called when async command result received, used by 3: other


		const char * confDbName;
		const char * histDbName;
		const char * dbHostIP;
		const char * dbUserName;
		const char * dbPassword;
		const char * dbPort;

		uint32_t channel;	// rtdb database id
		uint16_t beacon_port;	// xhpaxos broadcasting port
		uint16_t service_port;	// xhpaxos service port

		uint16_t version;	// xhpaxos version
		uint8_t cluster_server_cnt; // number of servers in this cluster, it must have more than half of servers online for cluster to operate
		uint8_t data_copy; // rtdb data replication count
		char dispatcher_type; // dispatcher_type; 1: rtdb server, 2: RTP server, 3: other, including proxy, pc clients
	};`

**Comment by saudet** [comment]
Someone needs to add support for that in JavaCPP.

**Comment by bruceywave** [comment]
I found that when I call javacpp functions from multithread, javacpp does not use thread local, that is why javacpp is slow

**Comment by saudet** [comment]
If I understand correctly, what you're looking for is `NO_JNI_DETACH_THREAD`, see https://github.com/bytedeco/javacpp/issues/143#issuecomment-262138799.

**Comment by saudet** [comment]
> I found that when I call javacpp functions from multithread, javacpp does not use thread local, that is why javacpp is slow

This has been fixed in pull #489 and released with JavaCPP 1.5.6. Enjoy!
